python/lsst/summit/utils/consdbClient.py

```diff
@@ -0,0 +1,412 @@
+    @staticmethod
+    def compute_flexible_metadata_table_name(instrument: str, obs_type: str) -> str:
+        """Compute the name of a flexible metadata table.
+
+        Each instrument and observation type made with that instrument can
+        have a flexible metadata table.  This function is useful when
+        issuing SQL queries, and it avoids a round-trip to the server.
+
+        Parameters
+        ----------
+        instrument: `str`
+            Name of the instrument (e.g. ``LATISS``).
+        obs_type: `str`
+            Name of the observation type (e.g. ``Exposure``).
+
+        Returns
+        -------
+        table_name: `str`
+            Name of the appropriate flexible metadata table.
+
+        """
+        return f"cdb_{instrument}.{obs_type}_flexdata"
+
+    def add_flexible_metadata_key(
+        self,
+        instrument: str,
+        obs_type: str,
+        key: str,
+        dtype: str,
+        doc: str,
+        unit: str = None,
+        ucd: str = None,
+    ) -> requests.Response:
+        """Add a key to a flexible metadata table.
+
+        Parameters
+        ----------
+        instrument: `str`
+            Name of the instrument (e.g. ``LATISS``).
+        obs_type: `str`
+            Name of the observation type (e.g. ``Exposure``).
+        key: `str`
+            Name of the key to be added (must not already exist).
+        dtype: `str`
+            One of ``bool``, ``int``, ``float``, or ``str``.
+        doc: `str`
+            Documentation string for the key.
+        unit: `str`, optional
+            Unit for the value. Should be from the IVOA
+            (https://www.ivoa.net/documents/VOUnits/) or astropy.
+        ucd: `str`, optional
+            IVOA Unified Content Descriptor
+            (https://www.ivoa.net/documents/UCD1+/).
+
+        Returns
+        -------
+        response: `requests.Response`
+            HTTP response from the server, with 200 status for success.
+        """
+        data = {"key": key, "dtype": dtype, "doc": doc}
+        if unit is not None:
+            data["unit"] = unit
+        if ucd is not None:
+            data["ucd"] = ucd
+        url = _urljoin(self.url, "flex", quote(instrument), quote(obs_type), "addkey")
+        return self._handle_post(url, data)
+
+    def get_flexible_metadata_keys(self, instrument: str, obs_type: str) -> dict[str, list[str | None]]:
+        """Retrieve the valid keys for a flexible metadata table.
+
+        Parameters
+        ----------
+        instrument: `str`
+            Name of the instrument (e.g. ``LATISS``).
+        obs_type: `str`
+            Name of the observation type (e.g. ``Exposure``).
+
+        Returns
+        -------
+        key_info: `list` [ `list` [ `str` | `None` ] ]
+            List of string lists, one for each key. Each string list contains
+            the name of the key, a data type string, and a documentation
+            string.  A unit string and UCD string may also be present.
```

Comment:
This makes me wonder if some simple dataclasses might simplify the interfaces. Perhaps we can add that later if it looks helpful though, but any time I see lists of lists with sub lists of variable length it gives me pause, as that can be quite clunky for any downstream code to use.

Reply:
Having seen the `schema()` function below, maybe at least coercing these two things to always have the same shape by adding empty strings when the types or units are not there? (And in the `schema()` function, making that be the same length, as that doesn't look like it carries a UCD string ever?

Reply:
Added a dataclass for flexible metadata, but `schema()` is native Postgres so can never have units or UCDs and has a larger dtype vocabulary, as mentioned in the Note, so I don't want it to look too similar.